### PR TITLE
validation: add more test cases for read-only paths tests

### PR DESCRIPTION
**Pull request by dongsupark**
Split out the test function into `checkReadonlyPaths()`, and add more cases for read-only paths like subdirectory, file under subdirectory, and directory under subdirectory.

Test inside container should return error if a relative path is given for read-only paths.

Deliberately create an invalid read-only symlink that points out of the container, to see if the test fails inside the container.

Create read-only block device, char device, and fifo, to check if they are read-only as expected.

### CODE REVIEW

File: validation/linux_readonly_paths.go [hunk 1]
```diff
@@ -28,13 +45,130 @@ func main() {
...
+}
+
+func checkReadonlyRelPaths() error {
+	g, err := util.GetDefaultGenerator()
+	if err != nil {
+		return err
+	}
+
+	// Deliberately set a relative path to be read-only, and expect an error
+	readonlyRelPath := "/etc/../readonly-relpath"
```
Comment (q384566678) [inline validation/linux_readonly_paths.go]: Is it a relative path? I think it equal to `/readonly-relpath`. It's a absolute path.

Comment (dongsupark) [reply validation/linux_readonly_paths.go]: @q384566678 Fixed the relative path.
Also fixed the same one in https://github.com/opencontainers/runtime-tools/pull/643.

Comment (q384566678) [reply validation/linux_readonly_paths.go]: Why not  use `readonlyRelPath := "readonly-relpath"` ?

Comment (dongsupark) [reply validation/linux_readonly_paths.go]: @q384566678 Done. Also done in #643.

**Comment by q384566678** [comment]
LGTM

**Pull request merged**